### DISCUSSION TITLE
Swagger UI support for OAuth2 authorization code flow

### DISCUSSION THREAD

**Discussion by jtv8** [fastapi-users/fastapi-users#949] [Ideas]
Hi!

Following on from #946, I'd like to ask what changes might be acceptable to support my use case of enabling the use of the authorization code flow in the Swagger UI.

From my point of view, I could implement everything in my downstream application by simply broadening the types that are acceptable to the `scheme` attribute:

```
class BearerTransport(Transport):
    scheme: OAuth2PasswordBearer
```

to:

```
class BearerTransport(Transport):
    scheme: OAuth2
```

That way I could override the `scheme` attribute to be an object of type `OAuth2AuthorizationCodeBearer`, and implement the necessary endpoints manually in my own application.

Of course, this would be neater still if we could have an intermediate class between `Transport` and `BearerTransport`, e.g.:

```
class BaseBearerTransport(Transport):
    scheme: OAuth2

    def __init__(self, scheme: OAuth2):
        self.scheme = scheme

...


class BearerTransport(BaseBearerTransport):

    def __init__(self, tokenUrl: str):
        super().__init__(OAuth2PasswordBearer(tokenUrl, auto_error=False))

...
```

Alternatively, do you have any other suggestions for enabling the use of the Swagger UI for this flow? It really is an incredibly useful tool for my team and we'd like to find a way to make use of it.

**Comment by frankie567** [comment]
Hmm, I see! Well, I kinda like your solution of having a generic `BaseBearerTransport` accepting any `OAuth2` scheme and make `BearerTransport` inherit from it. If you make a PR with this approach, I don't see any inconvenient with this 🙂
